### PR TITLE
Josh.wang/track pre checkout click event

### PR DESCRIPTION
**Pull request by wangjoshuah**

### CODE REVIEW

File: Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK Advanced Sample/AppDelegate.m [hunk 1]
```diff
@@ -43,13 +43,13 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     self.userId = [[NSUserDefaults standardUserDefaults] stringForKey:@"optlyUserId"];
     if (!self.userId) {
         // [OPTLY - Doc] Add user here for local testing
-        self.userId = @"";
+        self.userId = @"default_test_user";
```
Comment (mikeproeng37) [inline Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK Advanced Sample/AppDelegate.m]: nit: revert these changes

File: Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK Advanced Sample/PreCheckoutViewController.m [hunk 2]
```diff
@@ -168,6 +179,7 @@ -(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *
             }
             break;
         case UITableViewSectionContinue:
+            [self.optlyClient track:@"pre_checkout_cta_clicked" userId:self.userId attributes:self.userAttributes];
```
Comment (mikeproeng37) [inline Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK Advanced Sample/PreCheckoutViewController.m:182]: You can reuse the experiments I use for activation and track calls with them. They are already set up with audiences and groups. Also instead of hard-coding the experiment key, pass it in via params so you just need this one code change and you can write several test cases against it like I did for activation 

- cart_cta_with_audience
- cart_cta_no_audience
- grouped_cart_cta_experiment
- grouped_cart_cta_experiment_whitelist

**Review by mikeproeng37 — CHANGES_REQUESTED** [review]

File: Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK Advanced Sample/PreCheckoutViewController.m (hunk 2 above)
Comment (wangjoshuah) [reply Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK Advanced Sample/PreCheckoutViewController.m:182]: we're already reusing those by simply walking through the app. this is for tracking an event. the event is already linked to those 4 experiments.

Comment (mikeproeng37) [reply Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK Advanced Sample/PreCheckoutViewController.m:182]: Ah you're right. Ignore everything I said :)

**Review by mikeproeng37 — APPROVED** [review]
LGTM
